lib/chromic_pdf/pdf/browser/channel_state.ex

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule ChromicPDF.Browser.ChannelState do
+  @moduledoc false
+
+  alias ChromicPDF.Connection.JsonRPC
+  alias ChromicPDF.Protocol
+
+  @enforce_keys [:dispatch, :protocols]
+  defstruct [:dispatch, :protocols]
+
+  @type t :: %__MODULE__{dispatch: Protocol.dispatch(), protocols: [Protocol.t()]}
+
+  @spec new(Protocol.dispatch()) :: t()
+  def new(dispatch) do
+    %__MODULE__{dispatch: dispatch, protocols: []}
```

Comment:
tbh, I don't know what `dispatch` is even for. I only find one reference in `Protocol.advance/2` here, which seems to me to be ignored everywhere, right? 

If so wouldn't it be nicer if this module would be called `ProtocolExecution` and the state would be an array of them (with `defstruct [:protocol, :from, :ref]`) ? Some bits would move back to the channel, but I think this could work nicely still.

Reply:
`dispatch` is used in `Protocol.advance/2` when it processes `:call` steps. It allows the protocols to send messages (as a side-effect). I can refactor that to instead return a list of messages instead?

I'm not sure about `ProtocolExecution` still, or maybe I don't fully understand what you're saying: If it's just about making the triple in `ChannelState` a map/struct, I'm all for it, but I would prefer not to introduce either `from` or `ref` into `Protocol.t`

Reply:
how about `ChannelTask` or `ChannelOperation`?

Reply:
both work I guess... Slightly in favour of `ChannelTask`

>  I can refactor that to instead return a list of messages instead? 

Seems like a nice choice.